### ISSUE TITLE
Support for manipulating response body in Python PDK

### ISSUE THREAD

**Issue by Hicham-Chahboune** [Kong/kong-python-pdk#128]
Hello,

I'm developing a Kong plugin using the Python PDK. One of the requirements of my plugin is to manipulate the response body, similar to how it's done in the Lua PDK with ngx.arg[1] during the body_filter phase. (or response.set_raw_body)

However, I've noticed that the Python PDK doesn't seem to support the body_filter phase or provide a direct way to interact with the response body in chunks

**Comment by fffonion** [comment]
Hi @Hicham-Chahboune ,  `response.set_raw_body` is not supported in external plugins, however you can consider using `response.exit` instead.

**Comment by Hicham-Chahboune** [comment]
Thank you for your response. The issue with "response.exit" is that it interrupts the execution process of plugins. When a plugin calls "response.exit," it prevents the process from passing to other plugins

**Comment by fffonion** [comment]
@Hicham-Chahboune That is correct. Unfortunately that's the limitation we have.

**Comment by Hicham-Chahboune** [comment]
And with Lua, is it possible to change the content of the body? I tried to assign a new value to **ngx.arg[1]**, but it doesn't work. When I manipulate the content without changing it, it works (for example, using **ngx.arg[1] = string.upper(body)** works). However, when I try to assign a new variable like newbody, it does not work (for example, using `ngx.arg[1] = newbody`)

**Comment by fffonion** [comment]
yes in lua it's supported, the more detailed documentation is at openresty side (ref: https://github.com/openresty/lua-nginx-module#body_filter_by_lua_block)

**Comment by Hicham-Chahboune** [comment]
thank you for your support, here is how i solved the problem.
`
function PLUGIN:body_filter(conf)
  PLUGIN.super.body_filter(self)
  
  local body = ngx.arg[1]
  local eof = ngx.arg[2]
  
  if not ngx.ctx.buffered_body then
    ngx.ctx.buffered_body = {}
  end
  table.insert(ngx.ctx.buffered_body, body)
  
  if eof then
    local complete_body = table.concat(ngx.ctx.buffered_body)
    
    local new_body = "<soap:Envelope>3<soap:Envelope>"

    ngx.arg[1] = new_body
    ngx.header["Content-Length"] = tostring(#new_body)

  else
    ngx.arg[1] = nil
  end
end

function PLUGIN:header_filter(conf)
  PLUGIN.super.header_filter(self)
  ngx.header["Content-Length"] = nil
end
`

**Comment by ludovic-pourrat** [comment]
Any updates or chance to update the response ? The exit is not suitable and does not work in the context of the AI proxy.

**Comment by Robyt96** [comment]
I'm also trying to integrate some custom plugin with the AI proxy and being able to use the body_filter function would be useful.
